ledger/participant-integration-api/src/main/scala/platform/store/dao/events/BufferedTransactionsReader.scala

```diff
@@ -59,23 +58,23 @@ private[events] class BufferedTransactionsReader(
       startExclusive: Offset,
       endInclusive: Offset,
       filter: FilterRelation,
+      wildcardParties: Set[Party],
       eventProjectionProperties: EventProjectionProperties,
   )(implicit loggingContext: LoggingContext): Source[(Offset, GetTransactionsResponse), NotUsed] = {
-    val (parties, partiesTemplates) = filter.partition(_._2.isEmpty)
-    val wildcardParties = parties.keySet
-
-    val templatesParties = invertMapping(partiesTemplates)
-    val requestingParties = filter.keySet
-
+    val requestingParties = filter.values.flatten.toSet ++ wildcardParties
```

Comment:
how about adding this also to the case class for filter information? Then we would not have to do this calculation each and every time here, and also this computation would be removed from other places as well (for example in ACSReader we have the same computation)